packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -0,0 +1,269 @@
+                functionWithoutApiName: FunctionWithoutApiName,
+                functionWithApiName: FunctionWithApiName
+        ) {
+            var cancelToken : String
+            var apiName: String?
+            var options: RestOptions
+
+            try {
+                cancelToken = FlutterApiRequest.getCancelToken(request)
+                apiName = FlutterApiRequest.getApiPath(request)
+                options = FlutterApiRequest.getRestOptions(request)
+            } catch (e: Exception) {
+                FlutterApiError.postFlutterError(
+                        flutterResult,
+                        FlutterApiErrorMessage.AMPLIFY_REQUEST_MALFORMED.toString(),
+                        e)
+                return
+            }
+
+            try {
+                var operation: RestOperation?
+                if (apiName == null) {
+                    operation = functionWithoutApiName(options,
+                            Consumer { result ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareRestResponseResult(flutterResult, result, methodName) },
+                            Consumer { error ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                } else {
+                    operation = functionWithApiName(
+                            apiName,
+                            options,
+                            Consumer { result ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareRestResponseResult(flutterResult, result, methodName) },
+                            Consumer { error ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                }
+                OperationsManager.addOperation(cancelToken, operation!!)
+
+            } catch (e: Exception) {
+                prepareError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
```

Comment:
Can we make this a little cleaner to read?
Like `FlutterApiErrorMessage.getErrorForApi(methodName)`

It doesnt need to be Rest specific, and this function could be used by GraphQL in the future too since the enums are defined as follows:
```
AMPLIFY_API_QUERY_FAILED,
AMPLIFY_API_MUTATE_FAILED,
AMPLIFY_API_GET_FAILED,
AMPLIFY_API_POST_FAILED
```
Also consider just returning a `String` when calling this method and not having to add toString() at the end, which might be cleaner.